chain/x/oracle/types/exec_env.go

```diff
@@ -69,10 +69,10 @@ func NewPrepareEnv(req Request, maxRawRequests int64) *PrepareEnv {
 // AskExternalData implements Owasm ExecEnv interface.
 func (env *PrepareEnv) AskExternalData(eid int64, did int64, data []byte) error {
 	if int64(len(data)) > MaxRawRequestDataSize {
-		return api.ErrSpanExceededCapacity
+		return api.ErrSpanTooSmall
 	}
 	if int64(len(env.rawRequests)) >= env.maxRawRequests {
-		return api.ErrAskExternalDataExceed
+		return api.ErrWrongPeriodAction
```

Comment:
I think this error doesn't com from wrong period

Reply:
ErrTooManyExternalData ?

Reply:
Good catch